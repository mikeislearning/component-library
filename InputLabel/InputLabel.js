import React from 'react';
import { Text } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './InputLabel.styles';

export default class InputLabel extends React.Component {
  static propTypes = {
    hasTopMargin: PropTypes.bool,
    text: PropTypes.string,
  };

  static defaultProps = {
    hasTopMargin: false,
  };

  render() {
    const { hasTopMargin, text } = this.props;

    if (!text) {
      return null;
    }

    const style = [styles.label];
    if (hasTopMargin) style.push(styles.hasTopMargin);

    return <Text style={style}>{text}</Text>;
  }
}
