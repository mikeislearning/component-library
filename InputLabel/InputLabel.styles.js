import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  hasTopMargin: {
    marginTop: 30,
  },
  label: {
    color: colors.gray.medium,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 15,
    lineHeight: 16,
    marginBottom: 5,
  },
});

export default styles;
