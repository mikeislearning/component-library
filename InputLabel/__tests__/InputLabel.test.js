import React from 'react';
import { render } from 'react-native-testing-library';

import InputLabel from '../InputLabel';

describe('<InputLabel />', () => {
  it('renders correctly', () => {
    const label = render(<InputLabel text={'Input label'} />);
    expect(label).toMatchSnapshot();
  });

  it('renders with an extra top margin when needed', () => {
    const label = render(<InputLabel text={'Input label'} hasTopMargin={true} />);
    expect(label).toMatchSnapshot();
  });
});
