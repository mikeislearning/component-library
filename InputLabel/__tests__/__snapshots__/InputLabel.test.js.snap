// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputLabel /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#726E67",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 15,
        "lineHeight": 16,
        "marginBottom": 5,
      },
    ]
  }
>
  Input label
</Text>
`;

exports[`<InputLabel /> renders with an extra top margin when needed 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#726E67",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 15,
        "lineHeight": 16,
        "marginBottom": 5,
      },
      Object {
        "marginTop": 30,
      },
    ]
  }
>
  Input label
</Text>
`;
